### ISSUE TITLE
Gradient overflow issue when i try to train gpt2 with run_language_modeling in fp16 with 02. Any idea why that maybe happen?

### ISSUE THREAD

**Issue by niklaras** [huggingface/transformers#4692]
# ❓ Questions & Help

## Details

**A link to original question on Stack Overflow**:

**Comment by niklaras** [comment]
i get this message every 2000 steps. 
"Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 262144.0"

**Comment by BramVanroy** [comment]
Do you have the same issue with opt level O1? Using O2 is discouraged. The issue may also be related to PyTorch 1.5, so if switching to O1 does not help, try a previous PyTorch version. Note that it is highly likely that this is an AMP problem, not a transformers issue. Have a look here https://github.com/NVIDIA/apex/issues/318

**Comment by niklaras** [comment]
@BramVanroy  Yes i guess you 're right. It seems problem of the AMP. I asked and in APEX the same question and their answer was : 

"The loss scaler tries to increase the loss scaling factor after a threshold of successful steps was reached. In your case it seems that the scaling factor is being downgraded to the same value, so it should be fine."

So, according to that answer, is not a problem. But still i am not sure.

**Comment by BramVanroy** [comment]
I guess that skipping one step every 2000 steps is not a problem. You can monitor the loss, and as long as it seems to decrease normally, then you should be fine.
